src/components/ui/Modal.js

```diff
@@ -3,15 +3,18 @@ import ReactDom from 'react-dom';
 import { Fragment } from 'react';
 
 const ModalBackdrop = props => {
-  return <div className="modal-backdrop"></div>;
+  return <div onClick={props.onReset} className="modal-backdrop"></div>;
 };
 
 const overlay = document.getElementById('modal-overlay');
 
```

Comment:
Suggestion: try to not use document.getElementbyId as React is based on declarative programming. If we do imperative programming then it won't make sense as you would be directly using DOM instead of V-DOM.